lib/engine/step/buy_sell_par_shares.rb

```diff
@@ -188,6 +189,17 @@ def can_ipo_any?(entity)
         !bought? && @game.corporations.any? { |c| c.can_par?(entity) && can_buy?(entity, c.shares.first&.to_bundle) }
       end
 
+      def purchasable_companies_from_others(entity)
```

Comment:
return [] if bought? || !entity.cash.positive?...

@game.purchasable_companies.reject...

this is how you can consolidate can_buy_any_company?

Reply:
Have made an attempt now. Using name purchasable_companies in the step.